modules/dailymotionBidAdapter.js

```diff
@@ -159,11 +159,17 @@ export const spec = {
         // Vendor consent
         deepAccess(bidderRequest, 'gdprConsent.vendorData.vendor.consents.573') === true &&
         // Purposes
-        [1, 3, 4].every(v => deepAccess(bidderRequest, `gdprConsent.vendorData.purpose.consents.${v}`) === true) &&
+        [1, 3, 4].every(v =>
+          deepAccess(bidderRequest, `gdprConsent.vendorData.publisher.consents.${v}`) !== false &&
+          deepAccess(bidderRequest, `gdprConsent.vendorData.purpose.consents.${v}`) === true) &&
         // Flexible purposes
-        [2, 7, 9, 10].every(v =>
+        [2, 7, 9, 10].every(v => (
+          deepAccess(bidderRequest, `gdprConsent.vendorData.publisher.consents.${v}`) !== false ||
```

Comment:
The `vendorData.publisher.consents` is the [Publisher Purposes](https://github.com/InteractiveAdvertisingBureau/GDPR-Transparency-and-Consent-Framework/blob/master/TCFv2/IAB%20Tech%20Lab%20-%20Consent%20string%20and%20vendor%20list%20formats%20v2.md#publisher-purposes-transparency-and-consent) section, which is not what we're looking for.

We need to use the [Publisher *Restrictions*](https://github.com/InteractiveAdvertisingBureau/GDPR-Transparency-and-Consent-Framework/blob/master/TCFv2/IAB%20Tech%20Lab%20-%20Consent%20string%20and%20vendor%20list%20formats%20v2.md#what-are-publisher-restrictions) section to check if the publisher has indicated that we should switch legal basis for the purpose and vendor in question. Flexible purposes does *not* mean that we can look at one or the other.

Could it possibly be done the way it's done for `koblerBidAdapter.js`?
https://github.com/dailymotion-oss/Prebid.js/blob/217e494cb59313a6449beb9b953f6dfec58fd873/modules/koblerBidAdapter.js#L129-L136

Reply:
Thanks for noticing this !
I've trouble finding exact specification for the object structure, and can't get this objects filled with our integration on dailymotion.com.

`kobler` is expecting a value in `vendorData.publisher.restrictions.<purpose number>`, as you mentionned.

But beside, we have `quantcast` that expects the value in `vendorData.publisher.restrictions.<purpose number>.<vendor id>` : https://github.com/dailymotion-oss/Prebid.js/blob/217e494cb59313a6449beb9b953f6dfec58fd873/modules/quantcastBidAdapter.js#L84-L85 This seems more logical as restrictions can be given per vendor, but I can't find a reference for this.

In the meantime, I'm updating to use the same hierarchy as quantcast.

Reply:
Looking at the spec files, it looks like Kobler copied the code from Quantcast, but missed the vendor level:
https://github.com/prebid/Prebid.js/blob/cf98f85fda395b70bfe5816b2481b318417aac5c/test/spec/modules/koblerBidAdapter_spec.js#L21-L25
(Vendor 11 is Quantcast.)

Reply:
I have updated the publisher restriction implementation and tests following our discussion, could you have another look please ?